src/sentry/tasks/relocation.py

```diff
@@ -95,5 +137,333 @@ def uploading_complete(uuid: str) -> None:
+                    json.dumps(DEFAULT_CRYPTO_KEY_VERSION).encode("utf-8"),
+                )
+                decryptor = Fernet(plaintext_data_encryption_key)
+                json_data = decryptor.decrypt(unwrapped.encrypted_json_blob).decode("utf-8")
+            except Exception:
+                return fail_relocation(
+                    relocation,
+                    OrderedTask.PREPROCESSING_SCAN,
+                    ERR_PREPROCESSING_DECRYPTION,
+                )
+
+            # Grab usernames and org slugs from the JSON data.
+            usernames = []
+            org_slugs = []
+            try:
+                for json_model in json.loads(json_data):
+                    model_name = NormalizedModelName(json_model["model"])
+                    if get_model(model_name) == Organization:
+                        org_slugs.append(json_model["fields"]["slug"])
+                        # TODO(getsentry/team-ospo#190): Validate slug using regex, so that we can
+                        # fail early on obviously invalid slugs. Also keeps the database `JSONField`
+                        # from ballooning on bad input.
+                    if get_model(model_name) == User:
+                        usernames.append(json_model["fields"]["username"])
+                        # TODO(getsentry/team-ospo#190): Validate username using regex, so that we
+                        # can fail early on obviously invalid usernames. Also keeps the database
+                        # `JSONField` from ballooning on bad input.
+            except KeyError:
+                return fail_relocation(
+                    relocation, OrderedTask.PREPROCESSING_SCAN, ERR_PREPROCESSING_INVALID_JSON
+                )
+
+            # Ensure that the data is reasonable and within our set bounds before we start on the
+            # next task.
+            if len(usernames) == 0:
+                return fail_relocation(
+                    relocation,
+                    OrderedTask.PREPROCESSING_SCAN,
+                    ERR_PREPROCESSING_NO_USERS,
+                )
+            if len(usernames) > MAX_USERS_PER_RELOCATION:
+                return fail_relocation(
+                    relocation,
+                    OrderedTask.PREPROCESSING_SCAN,
+                    ERR_PREPROCESSING_TOO_MANY_USERS.substitute(count=len(usernames)),
+                )
+            if len(org_slugs) == 0:
+                return fail_relocation(
+                    relocation,
+                    OrderedTask.PREPROCESSING_SCAN,
+                    ERR_PREPROCESSING_NO_ORGS,
+                )
+            if len(org_slugs) > MAX_ORGS_PER_RELOCATION:
+                return fail_relocation(
+                    relocation,
+                    OrderedTask.PREPROCESSING_SCAN,
+                    ERR_PREPROCESSING_TOO_MANY_ORGS.substitute(count=len(org_slugs)),
+                )
+            missing_org_slugs = set(relocation.want_org_slugs) - set(org_slugs)
```

Comment:
Where is `want_org_slugs` coming from? Not seeing logic for it anywhere

Reply:
Set when the `Relocation` model is created via the `POST /relocation/` endpoint, see: https://github.com/getsentry/sentry/pull/59161